Updated custom linker file option in lm_run.py and makefile for TEST=…

amnafayyaz-lm opened the pull request:
…all command.
**Issue** : While running all the tests (directed and random both) simultaneously with TEST=all in the command , certain tests failed due to linker files issue.
**Resolved** : For  each test now it checks if there's custom linker file for that test then it uses that linker file otherwise the default one. This has been added in the lm_run.py.

haroonshafique-lm reviewed (changes requested):
@amnafayyaz-lm I have highlighted some redundant code that should be removed. Your logic seems great. 
However, the logic you have implemented can be applied using reduced lines. In all three functions [`gcc_compile`, `run_assembly` and `run_c`], you generate the path for specific linker file and then use two if else blocks within each function. You can do something like this 

```
cust_linker_path = <your logic here>
if os.path.exists(cust_linker_path):
  linker_path = cust_linker_path
```  
And the remaining code will work fine without using double if else blocks in functions.

amnafayyaz-lm commented:
Changes updated as mentioned in the review. Removed redundant code, updated the asm path, removed if else from main function and removed the multiple if else logic.

haroonshafique-lm commented on integration_files/SweRV_EH1/riscv_dv_extension/testlist.yaml:
```diff
@@ -1,4 +1,4 @@
-# Copyright Google LLC
+ sc# Copyright Google LLC
```
@amnafayyaz-lm please remove `sc` from first line.

haroonshafique-lm reviewed (approved):
Approved
